downstairs/src/lib.rs

```diff
@@ -1717,8 +1749,6 @@ where
     // continually locking the downstairs, cache the result here.
     let lossy = ads.lock().await.lossy;
 
-    tokio::pin!(dw_task);
```

Comment:
Curious, why no more `pin!`?  Is it because `fw` is no longer a mutex?

Reply:
The pin was never actually necessary, as far as I can tell; this diff removes it on `main`:
```diff
diff --git a/downstairs/src/lib.rs b/downstairs/src/lib.rs
index 427fdbd..76b0d95 100644
--- a/downstairs/src/lib.rs
+++ b/downstairs/src/lib.rs
@@ -1687,7 +1687,7 @@ where
      * function to take over and handle a reconnect or a new upstairs
      * takeover.
      */
-    let dw_task = {
+    let mut dw_task = {
         let adc = ads.clone();
         let fwc = fw.clone();
         tokio::spawn(async move {
@@ -1697,7 +1697,7 @@ where
 
     let (message_channel_tx, mut message_channel_rx) =
         mpsc::channel(MAX_ACTIVE_COUNT + 50);
-    let pf_task = {
+    let mut pf_task = {
         let adc = ads.clone();
         let tx = job_channel_tx.clone();
         let fwc = fw.clone();
@@ -1717,8 +1717,6 @@ where
     // continually locking the downstairs, cache the result here.
     let lossy = ads.lock().await.lossy;
 
-    tokio::pin!(dw_task);
-    tokio::pin!(pf_task);
     loop {
         tokio::select! {
             e = &mut dw_task => {
```

I guess pinning let us borrow the tasks mutably in the `select!`, which we could also do by just... making them mutable?